src/components/ArticlePage.jsx

```diff
@@ -22,24 +32,46 @@ export default function ArticlePage() {
                 setError("Failed to load article");
                 setIsLoading(false);
             });
-        }, [article_id]);
-        
-        if (isLoading) return <div>Loading...</div>;
-        if (error) return <div>Error: {error}</div>;
+    }, [article_id]);
+
+
+    const handleVote = (num) => {
+        setVote(currentVote => {
+            const newVote = currentVote + num;
+            patchVotes(article_id, num)
+                .catch(err => {
+                    console.error("Error updating votes:", err);
+                    setError("Failed to update votes");
+                    setVote(currentVote); 
+                });
+            return newVote;
+        });
+    };
+
+    if (isLoading) return <div>Loading...</div>;
+    if (error) return <div>Error: {error}</div>;
```

Comment:
I'm glad this is here as when offline this is working well. 

This is optional, consider what happens when you do this. Currently if the page loads and lets say I lose connection and try to vote, the entire article is gone. At least if the page is loaded I can still read it. Perhaps you could use this state to display an error without emptying the whole page? - This is just a suggestion and not a direction to change the whole thing if you're happy with it.